elements/lisk-tree/src/sparse_merkle_tree/sparse_merkle_tree.ts

```diff
@@ -35,8 +43,108 @@ export class SparseMerkleTree {
 	public get db(): Database {
 		return this._db;
 	}
+
+	public async getNode(nodeHash: Buffer): Promise<TreeNode> {
+		const data = await this._hashToValueMap.get(nodeHash);
+
+		if (!data) {
+			throw new Error(`Hash does not exist in merkle tree: ${nodeHash.toString('hex')}`);
+		}
+
+		if (isLeaf(data)) {
+			const { key, value } = getLeafData(data, this.keyLength);
+			return new Leaf(key, value);
+		}
+
+		const { leftHash, rightHash } = getBranchData(data);
+
+		return new Branch(leftHash, rightHash);
+	}
+
+	public async update(key: Buffer, value: Buffer) {
+		if (!value) {
+			throw new Error('Value cannot be empty');
+		}
+
+		if (key.length !== this.keyLength) {
+			throw new Error(`Key is not equal to defined key length of ${this.keyLength}`);
+		}
+
+		const newLeaf = new Leaf(key, value);
+		const ancestorNodes: TreeNode[] = [];
+		let bottomNode: TreeNode = new Empty();
+		let currentNode = await this.getNode(this.rootHash);
+		const binaryKey = binaryExpansion(key, this.keyLength);
+		let h = 0;
+
+		while (!isLeaf(currentNode.hash)) {
+			// Append currentNode to ancestorNodes
+			ancestorNodes.push(currentNode);
+			const d = binaryKey[h];
+			if (+d === 0) {
```

Comment:
On other places in code for similar check we are doing string comparison. I don't have preferences but we should use one consistent approach. Don't you think typecasting and then comparison is unnecessary overhead?